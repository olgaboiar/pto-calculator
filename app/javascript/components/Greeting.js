import React from 'react'

class Description extends React.Component {
  render () {
    return (
      <div
        data-testid="greeting"
        >Welcome to PTO Calculator</div>
    )
  }
}

export default Description
