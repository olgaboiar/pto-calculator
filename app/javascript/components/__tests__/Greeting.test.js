import { render } from "enzyme";
import * as React from "react";
import Greeting from "../Greeting"

describe("Greeting.js", () => {
  it("renders correctly", () => {  
    const wrapper = render(
      <Greeting />,
    );
  
    expect(wrapper).toMatchSnapshot();
  });

  it("shows brief description", () => {  
    const wrapper = render(
      <Greeting />,
    );
  
    expect(wrapper.text()).toContain("Welcome to PTO Calculator");
  });
})
