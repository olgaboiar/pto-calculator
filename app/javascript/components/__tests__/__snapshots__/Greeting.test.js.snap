// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Greeting.js renders correctly 1`] = `
initialize {
  "0": Object {
    "attribs": Object {
      "data-testid": "greeting",
    },
    "children": Array [
      Object {
        "data": "Welcome to PTO Calculator",
        "next": null,
        "parent": [Circular],
        "prev": null,
        "type": "text",
      },
    ],
    "name": "div",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": null,
    "prev": null,
    "root": Object {
      "attribs": Object {},
      "children": Array [
        [Circular],
      ],
      "name": "root",
      "namespace": "http://www.w3.org/1999/xhtml",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
      "x-attribsNamespace": Object {},
      "x-attribsPrefix": Object {},
    },
    "type": "tag",
    "x-attribsNamespace": Object {
      "data-testid": undefined,
    },
    "x-attribsPrefix": Object {
      "data-testid": undefined,
    },
  },
  "_root": [Circular],
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "normalizeWhitespace": false,
    "withDomLvl1": true,
    "xml": false,
  },
}
`;
